driver/src/price_estimation/dexag/api.rs

```diff
@@ -0,0 +1,183 @@
+#![allow(dead_code)]
+
+use anyhow::{anyhow, Context, Result};
+use ethcontract::Address;
+use isahc::prelude::*;
+use serde::{de::Error as _, Deserialize, Deserializer};
+use serde_with::rust::display_fromstr;
+use std::time::Duration;
+use url::Url;
+
+#[cfg_attr(test, mockall::automock)]
+/// Parts of the dex.ag api.
+pub trait DexagApi {
+    /// https://docs.dex.ag/api/tokens
+    fn token_list_full(&self) -> Result<Vec<Token>>;
+    /// https://docs.dex.ag/api/price
+    fn price(
+        &self,
+        from: &str,
+        to: &str,
+        amount: EitherAmount,
+        dex: Dex,
+        discluded: &str,
```

Comment:
what does discluded mean? Looks like it's [non-standard english for excluded](https://www.yourdictionary.com/disclude), maybe we can add a comment explaining its function.

Reply:
Checked on the documentation. Looks like this itself should be of type Vec<Dex> ?

Reply:
As in the other comment I want this to be a thin wrapper around their api which is why I kept the same name as their api docs use and did not write documentation comments. Technically this is vector but we are not going to use it so I don't think its worth it to write more code for this case until it's needed. I could remove both the dex and discluded parameter then we don't need the enum either.

Reply:
In that case, let's please remove it. I'd prefer a concise but cleanly typed API and we can add things later as they come up.